### ISSUE TITLE
[ERROR] EOF when reading a line

### ISSUE THREAD

**Issue by aaronbwcollins** [lindegroup/autopkgr#441]
When installing the JSS Importer the GUI will freeze completely on "Adding default AutoPkg repo for JSSImporter". After I force quit and restart the app, the tool is installed. When I try to configure it, it says its connected yet never prompts to add the JDS. AutoPkgr version 1.3.1

**Comment by eahrold** [comment]
@aaronbwcollins did the https://github.com/autopkg/jss-recipes.git successfully get added?

Also, after setting the API username and password, if the status indicator is green, all you need to do is check the `Use Master JDS` check box. No further action is required. 

(The prompts discussed in the README are only for non JDS share points, either shared via AFP/SMB.)

**Comment by aaronbwcollins** [comment]
The .jss recieps do show, however i get a :
"A Python exception occurred during the execution of autopkg, see the system log for more details.

[ERROR] EOF when reading a line"
I also get a orange triagle next to any .jss recipe.

**Comment by eahrold** [comment]
If your not too far along in your config quit AutoPkgr, remove the RecipeRepos folder, then do 

'defaults delete com.github.autopkg RECIPE_REPOS' 

Relaunch AutoPkgr, re-add recipes repo and the jss repo.

**Comment by aaronbwcollins** [comment]
Went ahead and did that, no luck. After readding the repos, I encounterd the same error.

**Comment by eahrold** [comment]
What action causes the error to show up?

**Comment by aaronbwcollins** [comment]
Anytime I try to run a recipe.

**Comment by homebysix** [comment]
I've been able to reproduce the initial freeze on "Adding default jss-recipes repo" during JSSImporter installation: https://github.com/lindegroup/autopkgr/issues/443

**Comment by homebysix** [comment]
This fix will be part of 1.3.2. Thanks for reporting it!

**Comment by homebysix** [comment]
Hi @aaronbwcollins — related issue #443 is due to be fixed in 1.3.2.

Are you still having the problem every time you try to run a recipe? Can you run a recipe in Terminal (e.g. `autopkg run Firefox.pkg`)?

**Comment by aaronbwcollins** [comment]
autopkg run Firefox.pkg
Processing Firefox.pkg...

Nothing downloaded, packaged or imported.

On Wed, Sep 9, 2015 at 2:42 PM, Elliot Jordan <notifications@github.com>
wrote:

> Hi @aaronbwcollins <https://github.com/aaronbwcollins> — related issue
> #443 <https://github.com/lindegroup/autopkgr/issues/443> is due to be
> fixed in 1.3.2.
>
> Are you still having the problem every time you try to run a recipe? Can
> you run a recipe in Terminal (e.g. autopkg run Firefox.pkg)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lindegroup/autopkgr/issues/441#issuecomment-139054560>
> .
>



-- 
Aaron B.W. Collins
Production Coordinator
Kicks
Production Office: 415-503-9400
Production Office Fax: 413-513-9400
Cell: 831.419.6084

**Comment by homebysix** [comment]
@aaronbwcollins Great, now try this:

1. Make sure you've got verbose logging turned on. (Click on the AutoPkgr suitcase icon in the menu bar, then select "Verbose Logs.")
2. In AutoPkgr, find Firefox.pkg in the recipe list. Right-click on it, and choose "Run This Recipe Only."
3. If an error of any kind occurs, go to Console.app and filter for AutoPkgr. Anything interesting? Paste here (minus confidential info). If no error occurs, repeat with Firefox.jss.

I'm trying to determine whether this is specific to jss recipes, or if it happens for every recipe.

**Comment by smashism** [comment]
+1

**Comment by aaronbwcollins** [comment]
Haven't had the chance to do this yet, however Skype was the only one that
successful installed on the jss.

**Comment by smashism** [comment]
Okay so this is regarding the "EOF when reading a line" thing…

When I re-installed AutoPkgr on my machine and ran a few recipes for the first time, I got the error. I then went through the recipes individually by right-clicking and running them 1x1, and those worked fine. Now that I've done that and run the recipes again, I don't get the error. This leads me to believe it might have something to do with directory creation for new recipes/repos in the `~/Library/AutoPkg/Cache/` and `~/Library/AutoPkg/RecipeRepos/` maybe?

**Comment by homebysix** [comment]
On the latest 1.3.2-alpha branch, I can no longer reproduce the jss-recipes repo-add freeze, nor do I get an EOF error. I hope that means this issue is fixed.

@smashism, did you have any luck with the 1.3.2-alpha I sent?

**Comment by smashism** [comment]
Unfortunately, no. The above was posted regarding behavior in 1.3.2-alpha.

**Comment by eahrold** [comment]
So, I've got an idea which is a bit of a long shot. In 1.3.1, to solve a possibly related issues, we started setting the working directory of the autopkg execution to the user's $TMPDIR. 
Purge that directory and try running AutoPkgr again.
```
rm -R $TMPDIR/* ; rm -R $TMPDIR/.*
```

**Comment by aaronbwcollins** [comment]
No Joy still getting EOF error with python.

**Comment by homebysix** [comment]
OK, I actually was able to reproduce this today. Here's what I did:

- First, I added a new recipe override for the SizeUp.jss recipe.
- Then I relaunched AutoPkgr and checked the box for the newly-added override.
- I clicked "Run Recipes Now."
- The EOF error appeared in a dialog box in AutoPkgr:
    ![screen shot 2015-09-15 at 11 58 31 am](https://cloud.githubusercontent.com/assets/7801391/9886833/342b78ee-5ba1-11e5-937a-da432fcbfb54.png)

In system.log, I saw the following traceback:
```
Sep 15 11:49:42 sfo-dp02.jamf.airgc.net AutoPkgr[18328]: (FULL AUTOPKG TRACEBACK) 
	
	  File "/usr/local/bin/autopkg", line 1660, in <module>
	
	    sys.exit(main(sys.argv))
	  File "/usr/local/bin/autopkg", line 1654, in main
	
	    exit(subcommands[verb]['function'](argv))
	  File "/usr/local/bin/autopkg", line 1420, in run_recipes
	
	    search_github=make_suggestions)
	  File "/usr/local/bin/autopkg", line 346, in load_recipe
	
	    recipe = load_recipe(parent_id, [], recipe_dirs)
	  File "/usr/local/bin/autopkg", line 346, in load_recipe
	
	    recipe = load_recipe(parent_id, [], recipe_dirs)
	  File "/usr/local/bin/autopkg", line 332, in load_recipe
	
	    search_github=search_github)
	  File "/usr/local/bin/autopkg", line 248, in locate_recipe
	
	    % (indef_article, name))
	EOFError: EOF when reading a line
```

So it seems clear to me that AutoPkg is coming upon an identifier in the recipe_list.txt for which it can't find the corresponding recipe file in the search path.

I dove deeper and looked at my recipe_list.txt. It had 19 recipe identifiers listed. I verified that disabling and re-enabling them in AutoPkgr changed the recipe_list.txt file as expected.

I disabled all recipes in AutoPkgr, then enabled only Firefox.pkg and clicked "Run Recipes Now." It worked!

I disabled Firefox.pkg and re-enabled a few more of my overrides. It worked again. I re-enabled all but the most recent addition, and it continued working. Then I re-enabled the SizeUp.jss recipe, and it failed with EOF again. Gotcha!

And here's why the problem happened:

- :white_check_mark: My SizeUp.jss recipe overrides `com.github.jss-recipes.jss.SizeUp`. I have the jss-recipes repo added, so no problem there.
- :white_check_mark: The `com.github.jss-recipes.jss.SizeUp` recipe uses `com.github.homebysix.pkg.SizeUp` as a parent. I have the homebysix-recipes repo added, so no problem there either.
- :warning: The `com.github.homebysix.pkg.SizeUp` recipe uses `com.github.poundbangbash.eholtam-recipes.irradiatedsoftware.sizeup.download` as a parent. __Oops, I forgot to add eholtam-recipes.__

_Note that AutoPkgr only warns you if a parent recipe is missing, not if a "grandparent" recipe is missing._

So, if you find yourself getting EOF errors, please try these steps:

1. Open Terminal and try `autopkg run --recipe-list ~/Library/Application\ Support/AutoPkgr/recipe_list.txt`
2. If you are prompted to `Search GitHub AutoPkg repos for...` then verify that you've added the repo for the identifier listed there. Repeat steps 1-2 until no error occurs.
3. If the recipes run fine in Terminal, then return to AutoPkgr and uncheck all your recipes. Inspect the recipe_list.txt file to make sure it's empty.
4. Check to make sure each recipe's full chain of parent recipes is present as you add them back in one at a time. Then try a full run again.

@aaronbwcollins Could you try steps 1-2 above (and 3-4 if needed) and let me know how it goes?

**Comment by eahrold** [comment]
One note about the solution. AutoPkg shouldn't "make suggestions" to add repos when a run is invoked using a `recipe-list.txt` file ([if there's more than one recipe in that file](https://github.com/autopkg/autopkg/blob/master/Code/autopkg#L1407-L1411))

The best way to get prompted to add the appropriate recipe-repos is by running a single recipe.

**Comment by benmartel** [comment]
I am not sure that I have been having the issue for some time but that seemed to work good for me ran with no problems.

**Comment by aaronbwcollins** [comment]
@homebysix Its worth mentioning I get the error when no Repos or Recipes are selected at all.
After adding the proper repos & recipes for 1Password, I was able to run the package. Its now clear to me that I ran into two different bugs. 
**The first** is the GUI freezing when running the install JSSImporter.
**The second** is the EOF, which is due to the repos/recipes missing when running a .jss recipe.
Knowing that, I was able to follow your instruction and ensure all .jss recipes had the corresponding repos/recipes.
Once the dependancies were in place it worked like a charm.
